packages/reactive-dict/reactive-dict.js

```diff
@@ -164,6 +164,24 @@ _.extend(ReactiveDict.prototype, {
     
   },
 
+  delete: function(key) {
+    var self    = this;
+    var oldKeys = self.keys;
+
+    _.each(oldKeys, function(value, oldKey) {
+      if (key == oldKey){
+        changed(self.keyDeps[oldKey]);
+        changed(self.keyValueDeps[oldKey][value]);
+        changed(self.keyValueDeps[oldKey]['undefined']);
+        delete self.keys[oldKey]; // clean up
+      }
+    });
+
+    // trigger a change regardless if any matching key found...
+    self.allDeps.changed();
+
```

Comment:
The reasoning behind putting the `allDeps.changed` and *always* changing, was for more predictability for end-users, we may want to move it up. :smile: 

Let me know